.github/workflows/library.yml

```diff
@@ -10,6 +10,9 @@ on:
 name: Library Crates
 
 env:
+  AR: llvm-ar
```

Comment:
The library action should be run on host instead of inside vm. So seem it doesn't rely on llvm-ar/nasm/clang.
What I hope to achieve is:
1) all components to run on host should be compiled with standard stable toolchain.
2) components which may be run inside the TD env should be compiled with special nightly toolchain/llvm/nasm.

Reply:
> The library action should be run on host instead of inside vm. So seem it doesn't rely on llvm-ar/nasm/clang.

@jiangliu - tbh I'm not clear how the GHA scripts were running previously if CC wasn't set for all steps that run `cargo`.

Note that previously, I found this warning in the CI logs (which is no longer shown):

```
warning: Linking two modules of different data layouts: '' is 'e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128' whereas 'td_payload.52f1f334-cgu.9' is 'e-m:e-i64:64-f80:128-n8:16:32:64-S128'
##[warning]Linking two modules of different data layouts: '' is 'e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128' whereas 'td_payload.52f1f334-cgu.9' is 'e-m:e-i64:64-f80:128-n8:16:32:64-S128'
```

I also still cannot work out why the 2019 build is failing. What I could do is add an env check to the build script so that if a var like `TDSHIM_DISABLE_BUILD_CHECKS=1`, the `check_environment()` function isn't called?